tests/Sinch.Tests/Core/HttpTests.cs

```diff
@@ -191,5 +196,102 @@ public async Task AddOwnHeaders()
 
             _httpMessageHandlerMock.VerifyNoOutstandingExpectation();
         }
+
+        [Fact]
+        public async Task UnauthorizedAndNoSecondAuthCallIfExpiredHeaderIsNotPresent()
+        {
+            _tokenManagerMock.GetAuthToken(Arg.Any<bool>())
+                .Returns("first_token");
+
+            var uri = new Uri("http://sinch.com/items");
+
+            // first token expires
+            _httpMessageHandlerMock.Expect(HttpMethod.Get, uri.ToString())
+                .WithHeaders("Authorization", "Bearer first_token")
+                .Respond(HttpStatusCode.Unauthorized);
+
+            var httpClient = new HttpClient(_httpMessageHandlerMock);
+            var http = new Http(_tokenManagerMock, httpClient, null, new SnakeCaseNamingPolicy());
+            Func<Task<EmptyResponse>> op1 = () => http.Send<EmptyResponse>(uri, HttpMethod.Get);
+
+            await op1.Should().ThrowAsync<SinchApiException>();
+            // only one call, returning cached token, is performed
```

Comment:
This comment is misleading as we are checking what's happening on the network here, and a "call" may make one think about an HTTP call while the call to the tokenManagerMock is a function call.
```suggestion
            // only one function call, to read the token stored in cache, is performed
```
Anyway, this seems to be an edge case as it would be awkward to have a JWT in the cache, that is not expired but that the API will treat as invalid.